### PR TITLE
Allow to start FLP directly from the flow logs producer

### PR DESCRIPTION
**Pull request by jotak**
## Description

- New "InProcess" ingest stage
- New entry point to start the whole FLP in-process: pipeline.StartFLPInProcess
- Add some tests
- Also moved the code to start prometheus into a new function/package to avoid repetitions

The idea behind is to reduce some of the overhead when the agent producing flows, and the FLP workload are running on the same host: in that case, the network connection between them could be avoided, saving resources.

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [x] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: cmd/flowlogs-pipeline/main.go [hunk 1]
```diff
@@ -182,27 +181,8 @@ func run() {
...
-		Addr: addr,
-		// TLS clients must use TLS 1.2 or higher
-		TLSConfig: &tls.Config{
-			MinVersion: tls.VersionTLS12,
-		},
-	}
-	tlsConfig := cfg.MetricsSettings.TLS
-	go utils.StartPromServer(tlsConfig, promServer, !cfg.MetricsSettings.NoPanic, prometheus.DefaultGatherer.(*prometheus.Registry))
+	prometheus.SetGlobalMetricsSettings(&cfg.MetricsSettings)
+	promServer := prometheus.StartServerAsync(&cfg.MetricsSettings.PromConnectionInfo, nil)
```
[the remark this answers is not in the document]
Comment (jotak) [reply cmd/flowlogs-pipeline/main.go:185]: It's not only for prometheus pipeline output but also for the "operational metrics" ie. related to FLP health .. which doesn't mean it cannot be made optional, but I would say the default should still to have these metrics

[the remark this answers is not in the document]
Comment (jotak) [reply cmd/flowlogs-pipeline/main.go:185]: I can add a field `enableOperational` in MetricsSettings.
If it's set false, this "global" server would not be started, and it's still possible to start per-promencode stage servers.
However if it's disabled and the user didn't configure per-promencode servers (which is optional), then they wouldn't get any metric ... so it creates more configuration skews....

[the remark this answers is not in the document]
Comment (jotak) [reply cmd/flowlogs-pipeline/main.go:185]: maybe `enableGlobalServer` would be a better name for the knob as it carries more explicitly that's it's neededif you don't configure per-stage server.

**Review by OlivierCazade — APPROVED** [review]

File: cmd/flowlogs-pipeline/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpinsonneau) [reply cmd/flowlogs-pipeline/main.go]: thanks !

**Pull request merged**